code/Player/Corpse.cs

```diff
@@ -73,7 +73,18 @@ public Corpse( Player player )
 		ApplyForceToBone( Player.LastDamage.Force, Player.GetHitboxBone( Player.LastDamage.HitboxIndex ) );
 
 		foreach ( var clothing in Player.Clothes.ToArray() )
-			clothing.SetParent( this, true );
+		{
+			if ( clothing.Model == Detective.Hat && (HitboxGroup)GetHitboxGroup( Player.LastDamage.HitboxIndex ) == HitboxGroup.Head )
+			{
+				clothing.Parent = null;
+				clothing.EnableAllCollisions = true;
+				clothing.PhysicsEnabled = true;
+				clothing.UsePhysicsCollision = true;
+				clothing.PhysicsGroup.AddVelocity( Player.LastDamage.Force.Length *  Vector3.Up );
```

Comment:
Maybe we should just wrap this up into a public method inside of `class Clothing`